File: phyton/Accessories.py
from PetShop import PetShop

class Accessories(PetShop):
    # konstruktor
    def __init__(self, id, name, price, stock, type, material, color,):
        super().__init__(id, name, price, stock)
        self.type = type
        self.material = material
        self.color = color
        
    # getter dan setter untuk type
    def setType(self, type):
        self.type = type
    
    def getType(self):
        return self.type
    
    # getter dan setter untuk material
    def setMaterial(self, material):
        self.material = material
    
    def getMaterial(self):
        return self.material
    
    # getter dan setter untuk color
    def setColor(self, color):
        self.color = color
    
    def getColor(self):
        return self.color

File: cpp/Accessories.cpp
#include <bits/stdc++.h>
#include "PetShop.cpp"

using namespace std;

class Accessories : public PetShop {
private:
    string type;
    string material;
    string color;
    
public:
    // konstruktor kosong
    Accessories()
    {
    }

    // konstruktor dengan parameter untuk mengisi atribut
    Accessories(string type, string material, string color)
    {
        this->type = type;
        this->material = material;
        this->color = color;
    }


    // setter type
    void setType(string type)
    {
        this->type = type;
    }

    // getter type
    string getType()
    {
        return type;
    }

    // setter material
    void setMaterial(string material)
    {
        this->material = material;
    }

    // getter material
    string getMaterial()
    {
        return material;
    }

    // setter color
    void setColor(string color)
    {
        this->color = color;
    }

    // getter color
    string getColor()
    {
        return color;
    }

    ~Accessories()
    {
    }
};
